forms-flow-web/src/components/Form/Item/View.js

```diff
@@ -409,25 +409,21 @@ const View = React.memo((props) => {
     };
   });
 
-  if (!user.role.some(el => el === STAFF_DESIGNER)) {
-    let formAccessInterval = null;
-    useEffect(() => {
-      formAccessInterval = setInterval(() => {
-        /* check formRef before calling function of formio */
-        if (formRef.current !== null) {
-          const formSupportedIdentityProviders = getFormSupportedIdentityProviders(
-            formRef.current?.formio, 
-            FORM_SUPPORTED_IDENTITY_PROVIDERS_FIELD_NAME, formAccessInterval);
-          if (Array.isArray(formSupportedIdentityProviders)) {
-            setHasFormAccess(hasUserAccessToForm(formSupportedIdentityProviders, user.username));
-          }
+  useEffect(() => {
+    if (user && user.role.some(el => el === STAFF_DESIGNER)) {
+      setHasFormAccess(true);
+    } else if (user && !user.role.some(el => el === STAFF_DESIGNER)) {
+      /* check formRef before calling function of formio */
+      if (formRef.current !== null) {
+        const formSupportedIdentityProviders = getFormSupportedIdentityProviders(
+          formRef.current?.formio, 
+          FORM_SUPPORTED_IDENTITY_PROVIDERS_FIELD_NAME, null);
+        if (Array.isArray(formSupportedIdentityProviders)) {
+          setHasFormAccess(hasUserAccessToForm(formSupportedIdentityProviders, user.username));
         }
-      }, 1000);
-      return () => {
-        clearInterval(formAccessInterval);
-      };
-    });
-  }
+      }
+    }
+  });
```

Comment:
I removed the `[user]` to make sure check for `formRef.currect` runs until it is available.